### ISSUE TITLE
shouldn't call setters while building existing model with isNewRecord=false

### ISSUE THREAD

**Issue by piyushkantm** [RobinBuschmann/sequelize-typescript#560]
This is somewhat related to https://github.com/RobinBuschmann/sequelize-typescript/issues/543.

I have this specific use case where i am caching user data in redis. and if cache is hit, i am rebuilding object from cached string instead of querying mysql (pretty standard use-case till now)

```
// Imports omitted...

export class User extends Model<User> {

    @Unique
    @PrimaryKey
    @AutoIncrement
    @Column(DataType.INTEGER.UNSIGNED)
    id: number;

    @Column(DataType.STRING)
    email: string;

    @Column({
        type: DataType.STRING,
        set: function (this: Super, value: string) {
            console.log("[User] Password Setter is called");
            this.setDataValue("password", cryptService.hashSync(value));
        }
    })
    password: string;

    // Lots of other properties... trimmed...
}
```

e.g. Record
```
{
  "id": 1,
  "email": "piyush@domain.com",
  "password": "$2b$10$1oPHsgsfd1Hf3WUJOpf2w9.fucz8jz.u7xmWXK82JJqysdfgtvbk9m"
}
```
So it is saved in redis as a string
```
{\"id\":1,\"email\":\"piyush@domain.com\",\"password\":\"$2b$10$1oPHsgsfd1Hf3WUJOpf2w9.fucz8jz.u7xmWXK82JJqysdfgtvbk9m\"}
```

So while fetching the string and rebuilding model (Or Hydrating in some paradigm), i do something like 

```
        const cachedUser = "{\"id\":1,\"email\":\"piyush@domain.com\",\"password\":\"$2b$10$1oPHsgsfd1Hf3WUJOpf2w9.fucz8jz.u7xmWXK82JJqysdfgtvbk9m\"}";
        const userModel = User.build(JSON.parse(cachedUser), {
            isNewRecord: false
        });
```

Now here comes the issue... Since this model is already in mysql, while building the object from JSON string, i expect two things:
1. When i try to update this object, it updates the existing row instead of trying to create a new one **Working. Thanks**
2. Setter is not called for password unless i operate on password property of this object. **This is not working**

In simpler terms **[User] Password Setter is called** this should not be logged in console unless i do either of the following:
```
        userModel.update({
            password: "NEW_PASS"
        });

        // OR

        userModel.password = "ANOTHER_NEW_PASS";
        userModel.update();
```
But unfortunately it does.. If it helps, i can setup a demo project but i think this is straight forward.

Thanks.

**Comment by RobinBuschmann** [comment]
Hey @piyushkantm, I couldn't find anything in the documentation of sequelize regarding when getters/setters should be called. Can you provide a link or something where the behaviour you've expected is documented?

**Comment by piyushkantm** [comment]
Apparently all i had to do was

```
        const userModel = User.build(JSON.parse(cachedUser), {
            isNewRecord: false,
            raw: true
        });
```

**Comment by piyushkantm** [comment]
@RobinBuschmann Sorry for late reply.. I got occupied in some other things and this issue was not on my most priority until now... 

After having a closer look at native sequelize library, i figured out what i was doing wrong...

**Comment by piyushkantm** [comment]
Just for the record. Here is an url pointing to Sequelize Getters/Setters doc:

http://docs.sequelizejs.com/manual/models-definition.html#getters--amp--setters

**Comment by piyushkantm** [comment]
By the way, I had the opportunity to look at the code of the library in an attempt to fix the issue which never existed. Hats off to the elegant code quality.
